pyrebase/pyrebase.py

```diff
@@ -87,18 +97,19 @@ def sign_in_with_email_and_password(self, email, password):
         return request_object.json()
 
     def create_custom_token(self, uid, additional_claims=None):
-        service_account_email = self.credentials.service_account_email
-        private_key = RSA.importKey(self.credentials._private_key_pkcs8_pem)
-        payload = {
-            "iss": service_account_email,
-            "sub": service_account_email,
-            "aud": "https://identitytoolkit.googleapis.com/google.identity.identitytoolkit.v1.IdentityToolkit",
-            "uid": uid
+        audience = "https://identitytoolkit.googleapis.com/google.identity.identitytoolkit.v1.IdentityToolkit"
+        claims = {
+            "uid": uid,
         }
         if additional_claims:
-            payload["claims"] = additional_claims
-        exp = datetime.timedelta(minutes=60)
-        return jwt.generate_jwt(payload, private_key, "RS256", exp)
+            claims.update(additional_claims)
+        credentials = jwt.Credentials.from_signing_credentials(
```

Comment:
Instead of creating a credentials class you don't need, you can actually use `self.credentials.signer` and `jwt.encode` directly.

Reply:
I originally had it set up the way you described. Then, I looked at the source of `jwt.Credentials.refresh` and realized it did the exact same thing with a more concise API. I also thought the current approach supported better separation of concerns—`pyrebase` doesn't have to worry about constructing a valid JWT payload in this case.

I admit `credentials.refresh(None)` is a little weird, but that seemed a small price to pay to avoid duplicating the `refresh` method.

What do you think?

Reply:
That's fine. :)